addons/editor-theme3/theme3.js

```diff
@@ -457,6 +457,52 @@ export default async function ({ addon, console, msg }) {
     Object.assign(originalColors, newColors);
   };
 
+  const updateMonitorColors = () => {
+    const allMonitors = addon.tab.redux.state.scratchGui.monitors.valueSeq();
+    const visibleMonitors = allMonitors.filter((monitor) => monitor.visible);
+    const monitorElements = document.querySelectorAll("[class*='monitor_monitor-container_']");
+    // The order of monitors in the Redux state and in the DOM is the same
+    visibleMonitors.forEach((monitor, i) => {
+      const opcodePrefix = monitor.opcode.split("_")[0];
+      let colorId =
+        {
+          sound: "sounds",
+          procedures: "more",
+        }[opcodePrefix] || opcodePrefix;
+      if (monitor.opcode === "data_listcontents") colorId = "data_lists";
+      let category = categories.find((category) => category.colorId === colorId);
+      if (!category) category = extensionsCategory;
+      for (const value of monitorElements[i].querySelectorAll(`
+        [class*="monitor_value_"],
+        [class*="monitor_large-value_"],
+        [class*="monitor_list-value_"]
+      `)) {
+        if (addon.settings.get("monitors")) {
+          value.style.backgroundColor = primaryColor(category);
+          value.style.color = isColoredTextMode() ? tertiaryColor(category) : uncoloredTextColor();
+          // Border color for list items
+          if (textMode() === "colorOnBlack") value.style.borderColor = "rgba(255, 255, 255, 0.15)";
+          else value.style.removeProperty("border-color");
+        } else {
+          value.style.backgroundColor = originalColors[category.colorId].primary;
+          value.style.color = originalColors.text;
+          value.style.removeProperty("border-color");
+        }
```

Comment:
Regarding this:

> If Scratch thinks it's in high-contrast mode and the setting is disabled, the updateMonitorColors() function should still run, but it would set monitors to the vanilla (original) colors, not the user-selected colors. What do you think?


Looks like the code already does this, but `originalColors` is defined to be the colors used by Scratch on page load (which misses the point). Would it hurt to hardcode these colors to be the **actual** original colors, which I assume is the original intention of this object?
Note: they were previously hardcoded, but [a commit which predates the Scratch theme selector](https://github.com/ScratchAddons/ScratchAddons/commit/b169b4461dd5755ad00bbdf7bfe3e2d81f7bd4b2#diff-58d2bc5ff8a91b2133df32c4c832575a79c3904cdc98cb7a3642158302794909R70) made the code shorter by getting the original colors from the Blockly object instead.

Reply:
> Also, I think the decision not to hardcode `originalColors` might cause bugs if during the same tab session, the user toggles the addon and changes the Scratch theme, as `originalColors` will no longer match `Blockly.Colours`

This was fixed in PR #6247. [This part](https://github.com/ScratchAddons/ScratchAddons/blob/c967343/addons/editor-theme3/theme3.js#L457-L460) of the code updates `originalColors` when the Scratch theme changes.

> I believe my suggestion is to use `Blockly.Colours` when appropriate (for example, if the addon is disabled) and rename `originalColors` to reflect it's the colors of the "Original" theme, which will be useful for this usecase. We could either hardcode the JSON object, or find a way to get it directly from the page somehow (we should also consider TurboWarp)

This wouldn't work - the addon works by changing `Blockly.Colours`, so it has to change them back when disabled, which is what the `originalColors` object is for. The monitor code could use a separate hardcoded object.